File: AUGUST-2025/15-08-2025/p342.cpp
class Solution {
public:
    bool isPowerOfFour(int n) {
        vector<int> arr = {1,4,16,64,256,1024,4096,16384,65536,262144,
                           1048576,4194304,16777216,67108864,268435456,1073741824};
        return find(arr.begin(), arr.end(), n) != arr.end();
    }
};

File: AUGUST-2025/15-08-2025/p342.kt
class Solution {
    fun isPowerOfFour(n: Int): Boolean {
        var n = n
        if (n and 0xFFFF == 0) {
            n = n shr 16
        }
        if (n and 0xFF == 0) {
            n = n shr 8
        }
        if (n and 0x0F == 0) {
            n = n shr 4
        }
        return n == 1 || n == 4 
    }
}